apps/src/music/views/MusicView.jsx

```diff
@@ -200,6 +198,11 @@ class UnconnectedMusicView extends React.Component {
         hasLoadedInitialSounds: false,
       });
       this.props.clearCallout();
+      this.musicBlocklyWorkspace.hideChaff();
+    }
+
+    if (this.props.appName !== 'music') {
+      this.musicBlocklyWorkspace.hideFlyout();
```

Comment:
What do you think about just calling `workspace.dispose()` whenever we change levels? Would that take care of both hiding the flyout and the chaff? I think we should re-initialize in `onLevelLoad` if we are still on a music page.

Reply:
Great question.

As far as each time we change levels, I'm slightly nervous about calling `dispose()` in the place we call `hideChaff()` (i.e. `prevProps.currentLevelIndex !== this.props.currentLevelIndex`), because I was earlier calling `hideFlyout()` there, and just one time I got to a music level and the flyout was gone.  I wasn't able to repro, but it seemed likely there is a subtle timing condition that would make disposal risky there.  

As far as just when we move to a non-`music` level, I just tried calling `dispose()` in the place we call `hideFlyout` (i.e. `this.props.appName !== 'music'`), but the workspace doesn't seem to come back after visiting a non-`music` level.  

As discussed [here](https://github.com/code-dot-org/code-dot-org/pull/55959#discussion_r1470261801), getting more crisp on lifecycle management would be great, but I think for now, I might stick with what seems to be working reliably.

Reply:
Quick follow-up:

> As far as just when we move to a non-music level, I just tried calling dispose() in the place we call hideFlyout (i.e. this.props.appName !== 'music'), but the workspace doesn't seem to come back after visiting a non-music level.

Oops, I had to set the workspace to `null` and then it worked.

`hideChaff` still offers some value in that it hides the unncessary UI elements immediately, and I remain nervous about using `dispose` in that location.

However, I found another case where `hideFlyout` didn't do everything: if a flyout was actually open, which `/s/music-intro-2024` offers in its functions level, the highlighted category title would persist into the non-music level.  

Switching to `dispose` resolved that issue, so I've prepped an update in https://github.com/code-dot-org/code-dot-org/pull/56139.

Reply:
> What do you think about just calling workspace.dispose() whenever we change levels?

After more consideration and testing, I updated https://github.com/code-dot-org/code-dot-org/pull/56139 to do what you suggested, and am calling `dispose()` whenever we are leaving a music level.  From a whole bunch of clicking, it seems to be working!